### PR TITLE
ITF: added fake peer synchronizer calls observable getters

### PR DESCRIPTION
**Pull request by MBoldyrev**
### Description of the Change

Methods to get observables on fake peer's synchronizer calls.

### Benefits

Additional handles for testing.

### Possible Drawbacks 

bugs.

### CODE REVIEW

File: test/framework/integration_framework/fake_peer/network/loader_grpc.hpp [hunk 1]
```diff
@@ -17,6 +18,20 @@ namespace integration_framework {
      public:
       explicit LoaderGrpc(const std::shared_ptr<FakePeer> &fake_peer);
 
+      bool sendBlockRequest(const std::string &dest_address,
```
Comment (Akvinikym) [inline test/framework/integration_framework/fake_peer/network/loader_grpc.hpp]: Please, add comments at least for return values of this and `sendBlocksRequest` methods. It is not immediately clear, what `bool` here and `size_t` there mean

Comment (igor-egorov) [reply test/framework/integration_framework/fake_peer/network/loader_grpc.hpp]: Please document all publically exposed methods.

File: test/framework/integration_framework/fake_peer/fake_peer.cpp [hunk 2]
```diff
@@ -282,5 +306,9 @@ namespace integration_framework {
                                                         request);
     }
 
+    void FakePeer::ensureInitialized() {
+      BOOST_VERIFY_MSG(initialized_, "Instance not initialized!");
```
Comment (igor-egorov) [inline test/framework/integration_framework/fake_peer/fake_peer.cpp]: Why we print here error message instead of just calling `initialize()` ?

Comment (MBoldyrev) [reply test/framework/integration_framework/fake_peer/fake_peer.cpp]: I would prefer the users of this interface to see explicitly where they operate on a not yet initialized object, and when they initialize it.

**Pull request merged**